public/app/core/components/AppChrome/AppChrome.tsx

```diff
@@ -31,6 +32,23 @@ export function AppChrome({ children }: Props) {
     [styles.contentChromeless]: state.chromeless,
   });
 
+  const handleMegaMenu = () => {
+    switch (state.megaMenu) {
+      case 'closed':
+        chrome.setMegaMenu('open');
+        break;
+      case 'open':
+        chrome.setMegaMenu('closed');
+        break;
+      case 'docked':
+        // on desktop, clicking the button when the menu is docked should close the menu
+        // on mobile, the docked menu is hidden, so clicking the button should open the menu
+        const isDesktop = window.innerWidth > theme.breakpoints.values.md;
+        isDesktop ? chrome.setMegaMenu('closed') : chrome.setMegaMenu('open');
+        break;
```

Comment:
not sure if introducing this extra logic is worth it for the pretty niche use case of switching between desktop and mobile widths... wdyt? 🤔

Reply:
Can you please explain this to me more? 

Are you saying this is handling if someone docks the mega menu on desktop, then resizes the window to the mobile breakpoint, then clicking the hamburger menu should do the 'correct' thing? Otherwise it gets into a weird state?

Reply:
yep exactly

- you're on desktop width, you dock the menu
- you drag the window to mobile width, the docked menu is now hidden
- **with** this logic, if you click the hamburger button the menu opens properly
- **without** this logic, if you click the hamburger button it first "closes" the menu so appears to do nothing. after that first click it behaves as normal.

Reply:
I think this is worth it to avoid getting in that funky state. While we do currently show docked nav on both ipad orientations, if that changes in the future it's important we handle this properly i think.

Reply:
another option i was thinking about was properly changing the underlying state of the docked menu to "closed" when you switch to mobile widths, but then if you dragged the screen back to desktop widths it wouldn't remember it was docked so that's also a bit funky. we can iterate 🤷